openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py

```diff
@@ -0,0 +1,129 @@
+    """
+    discussion_id = String(scope=Scope.settings, default=UNIQUE_ID, force_export=True)
+    display_name = String(
+        display_name=_("Display Name"),
+        help=_("Display name for this module"),
+        default="Discussion",
+        scope=Scope.settings
+    )
+    discussion_category = String(
+        display_name=_("Category"),
+        default=_("Week 1"),
+        help=_(
+            "A category name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    discussion_target = String(
+        display_name=_("Subcategory"),
+        default="Topic-Level Student-Visible Label",
+        help=_(
+            "A subcategory name for the discussion. "
+            "This name appears in the left pane of the discussion forum for the course."
+        ),
+        scope=Scope.settings
+    )
+    sort_key = String(scope=Scope.settings)
+
+    editable_fields = ["display_name", "discussion_category", "discussion_target"]
+
+    has_author_view = True  # Tells Studio to use author_view
+
+    @property
+    def course_key(self):
+        """
+        :return: int course id
+        """
+        return getattr(self.scope_ids.usage_id, 'course_key', None)
+
+    @property
+    def django_user(self):
+        """
+        Returns django user associated with user currently interacting
+        with the XBlock.
+        """
+        user_service = self.runtime.service(self, 'user')
+        if not user_service:
+            return None
+        return user_service._django_user  # pylint: disable=protected-access
+
+    def has_permission(self, permission):
+        """
+        Encapsulates lms specific functionality, as `has_permission` is not
+        importable outside of lms context, namely in tests.
+
+        :param user:
+        :param str permission: Permission
+        :rtype: bool
+        """
+        # normal import causes the xmodule_assets command to fail due to circular import - hence importing locally
+        from django_comment_client.permissions import has_permission  # pylint: disable=import-error
+
+        return has_permission(self.django_user, permission, self.course_key)
+
+    def student_view(self, context=None):
+        """
+        Renders student view for LMS.
+        """
+        fragment = Fragment()
+
+        course = self.runtime.modulestore.get_course(self.course_key)
+
+        context = {
+            'discussion_id': self.discussion_id,
+            'user': self.django_user,
+            'course': course,
+            'course_id': self.course_key,
+            # used as template condition: must be boolean
+            'can_create_thread': self.has_permission("create_thread"),
+            # used as html data: must be JSON
+            'can_create_comment': json.dumps(self.has_permission("create_comment")),
```

Comment:
These should be consistent, and you can just move the `json.dumps` calls into the template instead.